Keyboard/KeyboardViewController.swift

```diff
@@ -757,8 +757,8 @@ class KeyboardViewController: UIInputViewController {
     }
     
     func stringIsWhitespace(_ string: String?) -> Bool {
-        if string != nil {
-            for char in (string!).characters {
+        if let string = string {
```

Comment:
I'd probably use a guard here instead, or even change the method to only accept unwrapped strings.